lstm/evaluate.py

```diff
@@ -15,13 +16,20 @@ def summary(self):
         print("{:.2f}\t{:.2f}\t{:.2f}".format(self.precision, self.recall, self.f1))
 
 class Evaluator(object):
-    def __init__(self, results_file, neg_class = "None"):
-        self.results_file = results_file
+    def __init__(self, df, labels = ['E1 precedes E2', 'E2 precedes E1'], neg_class = "None"):
+        self.df = df
         self.neg_class = neg_class
         self.smoothing = 0.00001
-        
+        self.labels = labels
+
+    def micro(self):
+        return metrics.f1_score(self.df.Gold.values, self.df.Predicted.values, self.labels, average="micro")
+    
+    def macro(self):
+        return metrics.f1_score(self.df.Gold.values, self.df.Predicted.values, self.labels, average="macro")
+    
     def compute_class_score(self, class_label):
-        df = pd.read_csv(self.results_file, delimiter="\t")
+        df = self.df.copy(deep=True)
```

Comment:
Fair enough. I think I was just being overly cautious and never reviewed it again.